### PR TITLE
feat: eager loading of wasm in jest environment

### PR DESCRIPTION
**Pull request by mcfedr**
fixes https://github.com/nodejs/undici/issues/1235

Its pretty horrible, and but really need a solution to this problem.

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

**Review by ronag — APPROVED** [review]

**Comment by mcollina** [comment]
Can you add a simple test with Jest?

File: lib/client.js [hunk 1]
```diff
@@ -407,18 +407,21 @@ class Client extends Dispatcher {
...
 const EMPTY_BUF = Buffer.alloc(0)
 
+const llhttpSimdWasmData = process.env.JEST_WORKER_ID ? require('./llhttp/llhttp_simd.wasm.js') : undefined
+const llhttpWasmData = process.env.JEST_WORKER_ID ? require('./llhttp/llhttp.wasm.js') : undefined
+
 async function lazyllhttp () {
   let mod
   try {
-    mod = await WebAssembly.compile(Buffer.from(require('./llhttp/llhttp_simd.wasm.js'), 'base64'))
+    mod = await WebAssembly.compile(Buffer.from(llhttpSimdWasmData ?? require('./llhttp/llhttp_simd.wasm.js'), 'base64'))
```
Comment (ronag) [inline lib/client.js:416]:
```suggestion
    mod = await WebAssembly.compile(Buffer.from(llhttpSimdWasmData || require('./llhttp/llhttp_simd.wasm.js'), 'base64'))
```

File: lib/client.js [hunk 2]
```diff
@@ -407,18 +407,21 @@ class Client extends Dispatcher {
...
   } catch (e) {
     /* istanbul ignore next */
 
     // We could check if the error was caused by the simd option not
     // being enabled, but the occurring of this other error
     // * https://github.com/emscripten-core/emscripten/issues/11495
     // got me to remove that check to avoid breaking Node 12.
-    mod = await WebAssembly.compile(Buffer.from(require('./llhttp/llhttp.wasm.js'), 'base64'))
+    mod = await WebAssembly.compile(Buffer.from(llhttpWasmData ?? require('./llhttp/llhttp.wasm.js'), 'base64'))
```
Comment (ronag) [inline lib/client.js:424]:
```suggestion
    mod = await WebAssembly.compile(Buffer.from(llhttpWasmData || require('./llhttp/llhttp.wasm.js'), 'base64'))
```

**Review by ronag — CHANGES_REQUESTED** [review]
See suggestions. Needs to work on node 12.

File: package.json [hunk 3]
```diff
@@ -48,7 +48,7 @@
     "test": "npm run test:tap && npm run test:node-fetch && npm run test:fetch && npm run test:jest && tsd",
     "test:node-fetch": "node scripts/verifyVersion.js 16 || mocha test/node-fetch",
     "test:fetch": "node scripts/verifyVersion.js 16 || tap test/fetch/*.js",
-    "test:jest": "jest test/jest/test",
+    "test:jest": "jest test/jest/*.js",
```
Comment (mcfedr) [inline package.json:51]: @mcollina found there was a test file for jest, that had been disabled, and by changing this and it running, we actually see this error, and with this fix its resolved

**Review by mcollina — APPROVED** [review]
lgtm

**Review by ronag — APPROVED** [review]

File: lib/client.js [hunk 4]
```diff
@@ -408,6 +408,8 @@ const constants = require('./llhttp/constants')
 const EMPTY_BUF = Buffer.alloc(0)
 
 async function lazyllhttp () {
+  const llhttpWasmData = process.env.JEST_WORKER_ID ? require('./llhttp/llhttp.wasm.js') : undefined
```
Comment (ronag) [inline lib/client.js:411]: is this eager enough?

Comment (ronag) [inline lib/client.js:411]: Now I don't understand? What's the purpose? This isn't very eager.

**Review by ronag — APPROVED** [review]

**Comment by mcfedr** [comment]
I guess im done changing it, the last change got windows to find the tests, looks like wont run on macos because of my access level, but i've run it on macos myself

**Pull request merged**